src/Container/Container.php

```diff
@@ -0,0 +1,465 @@
+			$assocProvidedParameters[$key] = $value;
+		}
+
+		return array_replace($assocReflectionParameters, $assocProvidedParameters);
+	}
+
+	/**
+	 * Returns the name of function.
+	 *
+	 * @param \ReflectionParameter $parameter ReflectionParameter instance.
+	 *
+	 * @since 3.0.0
+	 *
+     * Note: Adapted from mako\framework.
+     *
+     * @see https://github.com/mako-framework/framework/blob/master/LICENSE
+	 *
+	 * @return string
+	 */
+	protected function getDeclaringFunction(\ReflectionParameter $parameter): string
+	{
+		$declaringFunction = $parameter->getDeclaringFunction();
+		$class = $parameter->getDeclaringClass();
+
+		if ($declaringFunction->isClosure()) {
+			return "Closure";
+		} elseif($class === null) {
+			return $declaringFunction->getName();
+		}
+
+		return $class->getName() . '::' . $declaringFunction->getName();
+
+	}
+
+	/**
+	 * Resolve a parameter.
+	 *
+	 * @param \ReflectionParameter  $parameter ReflectionParameter instance.
+	 * @param \ReflectionClass|null $class     ReflectionClass instance.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return mixed
+	 */
+	protected function resolveParameter(\ReflectionParameter $parameter, \ReflectionClass $class = null)
+	{
+		//Try to get the class name.
+		if ($parameterClass = $parameter->getClass() !== null) {
+			$parameterClassName = $parameterClass->getName();
+			if ($class !== null) {
+				$parameterClassName = $this->resolveContextualDependency($class->getName(), $parameterClassName);
+			}
+
+			return $this->get($parameterClassName);
+
+			//Detetmine Parameter has default value? yes, use it.
+		} elseif ($parameter->isDefaultValueAvailable()) {
+			return $parameter->getDefaultValue();
+		}
+
+		throw new \RuntimeException("Unable to resolve the parameter", 500);
+		
+	}
+
+	/**
+	 * Resolve parameters.
+	 *
+	 * @param array                 $reflectionParameters Reflection parameters.
+	 * @param array                 $providedParameters   Provided Parameters.
+	 * @param  \ReflectionClass|null $class                ReflectionClass instance.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return array
+	 */
+	protected function resolveParameters(array $reflectionParameters, array $providedParameters, \ReflectionClass $class = null): array
+	{
+		//Merge the parameter in to one array.
+		$parameters = $this->mergeParameters($reflectionParameters, $providedParameters);
+
+		foreach($parameters as $key => $parameter)
+		{
+			//Determine either the parameter instance of \ReflectionParameter?
+			if($parameter instanceof \ReflectionParameter)
+			{
+				$parameters[$key] = $this->resolveParameter($parameter, $class);
+			}
+		}
+
+		return array_values($parameters);
+	}
+
+	/**
+	 * Creates a class instance using closure.
+	 *
+	 * @param closure $factory      Closuare.
+	 * @param array   $parameters Constructor parameters.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return object
+	 */
+	public function closureFactory(\Closure $factory, $parameters)
```

Comment:
It should add the `array` type hint for `$parameters`.

Reply:
could you explain more, i thought you mean we can append typeHint to parameter array

Reply:
Consider following code snippets:

```php
...
public function closureFactory(\Closure $factory, array $parameters)
...
```